Source/MLX/DType.swift

```diff
@@ -201,16 +247,10 @@ public protocol ScalarOrArray {
     func asMLXArray(dtype: DType?) -> MLXArray
 }
 
-extension Int: ScalarOrArray {
-    public func asMLXArray(dtype: DType?) -> MLXArray {
-        // callers can use Int64() to get explicit .int64 behavior
-        MLXArray(Int32(self), dtype: dtype ?? .int32)
-    }
-}
-
 extension Double: ScalarOrArray {
     public func asMLXArray(dtype: DType?) -> MLXArray {
-        MLXArray(Float(self), dtype: dtype ?? .float32)
+        let dtype = dtype ?? .float32
+        return MLXArray(Float(self), dtype: dtype.isFloatingPoint ? dtype : .float32)
```

Comment:
If I pass an integer dtype here shouldn't it give me an integer array?

Like in Python if I do:

```
mx.array(1.0, mx.int32)
```

I get an `int32` array.

Reply:
It's likely I'm misunderstanding the purpose of these functions, so apologies in advance.

Reply:
This is for the case where you do:

```python
>>> mx.array(10.5) + 3
array(13.5, dtype=float32)
```

It is the equivalent of this:

```
array to_array(
    const ScalarOrArray& v,
    std::optional<Dtype> dtype /* = std::nullopt */) {
```